cmake/modules/FindPETSC.cmake

```diff
@@ -197,16 +197,28 @@ IF(NOT PETSC_PETSCVARIABLES MATCHES "-NOTFOUND")
   ENDFOREACH()
 ENDIF()
 
+IF(PETSC_WITH_MPIUNI)
+  #
+  # Workaround: Some distributions happen to not install petscvariables and
+  # we consequently might miss some essential include directories. Let's
+  # try at least to find the mpiuni include directory.
+  #
+  DEAL_II_FIND_PATH(PETSC_INCLUDE_DIR_MPIUNI mpi.h
+  HINTS ${PETSC_INCLUDE_DIR_COMMON} ${PETSC_INCLUDE_DIR_ARCH} ${_petsc_includes}
+  PATH_SUFFIXES mpiuni petsc/mpiuni
```

Comment:
Is there a risk that we find the global `mpi.h` if someone has an MPI installed on their system, but chosen to configure PETSc without it?